internal/test/fake_binary.go

```diff
@@ -0,0 +1,75 @@
+// Copyright 2021 Tetrate
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package test
+
+import (
+	"os"
+	"os/exec"
+	"path/filepath"
+	"runtime"
+	"testing"
+
+	"github.com/stretchr/testify/require"
+
+	"github.com/tetratelabs/func-e/internal/moreos"
+)
+
+var (
+	_, currentFilePath, _, _ = runtime.Caller(0)
+
+	// funcEGoModDir points to the directory where the func-e's go.mod resides.
```

Comment:
this is neat, but it isn't as isolated as I was hoping we could be. the variant here accomplishes the minimum dependencies, and will help us avoid bloat. For example, we shouldn't be adding massive deps like gopsutil into moreos. isolating helps avoid that https://github.com/tetratelabs/func-e/compare/fake-deux

Reply:
Yes, I was either do the copying or this. Was optimizing for minimal diff. But the sketch looks good, I'll accommodate that. Thanks, @codefromthecrypt.